recipe/meta.yaml

```diff
@@ -20,7 +20,7 @@ requirements:
   host:
     - python
     - pip
-    - hatchling
+    - uv-build
```

Comment:
Could you try moving this line to the `build:` section of requirements?

The `bad executable` error usually happens because an executable that was compiled for the target architecture is (attempted to) run on the build architecture. For osx-arm64, you are cross-compiling here and so the uv-build executable needs to be able to run on amd64.